BD New Design - Elements

shervElmi opened the pull request:
With this PR, we will add Buttons, Forms, Loaders, Modal, and Alerts styles of the new design.

### Elements Screenshots:
#### Alerts and Forms Elements
<kbd>
<img width="940" alt="image" src="https://user-images.githubusercontent.com/69119241/228669093-27a5402f-3066-45a9-a818-918b95cb6ef3.png">
</kbd>

### Buttons
<kbd>
<img width="701" alt="image" src="https://user-images.githubusercontent.com/69119241/228680404-4f7bd7be-8f54-4c71-8144-50b6770c11ad.png">
</kbd>

### Buttons with the "Button Style" option enabled
<kbd>
<img width="756" alt="image" src="https://user-images.githubusercontent.com/69119241/228680265-cee5439a-b363-4ad1-9696-56d641484b18.png">
</kbd>

shervElmi commented:
@stephywells could you please review this design PR first? thank you!

stephywells reviewed (changes requested):
I'm seeing a lot of things that don't look right. Can you go through and review this PR again? It would be helpful to include screenshots in the PR description too, so I know better what to look for.

Here are a few issues I'm seeing:
<img width="314" alt="Screenshot 2023-03-20 at 10 00 30 AM" src="https://user-images.githubusercontent.com/1116876/226398714-700e100b-123d-467f-aea3-2a36a98e58fb.png">

<img width="517" alt="Screenshot 2023-03-20 at 10 00 12 AM" src="https://user-images.githubusercontent.com/1116876/226398851-d1249cb6-b739-45b8-aeff-8047e3eaa64d.png">
<img width="217" alt="Screenshot 2023-03-20 at 9 57 22 AM" src="https://user-images.githubusercontent.com/1116876/226398881-90234a67-3bbc-4fdf-9891-82a0ab7ad4a7.png">

<img width="1248" alt="Screenshot 2023-03-20 at 9 53 30 AM" src="https://user-images.githubusercontent.com/1116876/226398917-770443ca-f412-4d7c-825b-89926dde82ca.png">

Missing field description styling:
<img width="604" alt="Screenshot 2023-03-20 at 9 53 36 AM" src="https://user-images.githubusercontent.com/1116876/226398953-5ba649c2-60c1-429e-9555-7f7102774b12.png">

shervElmi commented on includes/helpers/functions/templates-ui.php:
```diff
@@ -329,7 +329,7 @@ function wpbdp_main_links( $buttons = null ) {
 	if ( in_array( 'create', $buttons, true ) ) {
 		$html[] = '<a href="' . esc_url( wpbdp_url( 'submit_listing' ) ) . '" id="wpbdp-bar-submit-listing-button" class="button wpbdp-button">' .
-			esc_html__( 'Add Listing', 'business-directory-plugin' ) .
+			esc_html__( '+ Add Listing', 'business-directory-plugin' ) .
```
Hi Steph, thanks for the review.

Actually, these are basic styles and we'll focus on specific parts in the component PRs. For instance, in this PR, we only set basic button styles to avoid repetition.

shervElmi commented on assets/css/less/wpbdp.less:
```diff
@@ -447,37 +456,10 @@ form#wpbdmsearchform a.advanced-search-link {
 	clear: none;
 }
 
-@import "form-fields.less";
```
This file contains checkout form styles and other related elements that we will move to their respective component styles. As mentioned earlier, this PR only applies basic styles to elements, such as form inputs.

shervElmi replied:
For instance, the styles for submitting form fields can be found here:

[assets/css/less/wpbdp/components/submit-listing/listing-fields-section.less](https://github.com/Strategy11/business-directory-plugin/pull/295/files#diff-c4fe82c5553f9747721df0f0b96aa5a4e3efb7ea28901eee1ff1fe4f515c3852)

stephywells replied:
Thanks for moving that css. As I mentioned before, moving the css is fine but I don't want to merge any PRs that remove css without adding it back in.

shervElmi commented on assets/css/less/front-end/elements/alerts.less:
```diff
@@ -0,0 +1,30 @@
+	background-color: @bg-warning;
+	color: @color-text--60;
+	font-size: @font-size-alert;
+	line-height: @line-height-body;
```
I think it's okay to use the line-height property for specific elements like alerts.

shervElmi commented:
@stephywells I've updated the `forms.less` file and made other requested changes. Could you please review my PR again? Thank you.

shervElmi commented:
@stephywells, This is ready for review. Thanks in advance!

stephywells commented:
@shervElmi I'm still seeing most of the same issues from my previous screenshots. I still haven't seen screenshots from you to show me what to expect. Can you please focus on this PR and get it completed today?

shervElmi commented:
> @shervElmi I'm still seeing most of the same issues from my previous screenshots. I still haven't seen screenshots from you to show me what to expect. Can you please focus on this PR and get it completed today?

I've updated the description and added relevant screenshots to descriptions.

stephywells commented:
Thanks. Here's what I'm seeing now:

Clear on hover:
<img width="1237" alt="Screenshot 2023-03-30 at 9 52 20 AM" src="https://user-images.githubusercontent.com/1116876/228893806-839ec705-fa55-43d4-9de9-36b65001c487.png">

Buttons look like links until you hover:
<img width="317" alt="Screenshot 2023-03-30 at 9 52 28 AM" src="https://user-images.githubusercontent.com/1116876/228893808-9b0a36a7-7fc7-4caa-8b51-da3df9fea9a8.png">
<img width="404" alt="Screenshot 2023-03-30 at 9 52 46 AM" src="https://user-images.githubusercontent.com/1116876/228893810-2c260365-8c4b-4001-a407-3f35f8626684.png">

I understand you have other PRs that extend this, but I need all the elements you are working with here to either look the same or better. I don't want to merge anything that looks worse, even if a different PR adjusts it.

shervElmi commented:
> > Thanks. Here's what I'm seeing now:
> > Clear on hover: <img alt="Screenshot 2023-03-30 at 9 52 20 AM" width="1237" src="https://user-images.githubusercontent.com/1116876/228893806-839ec705-fa55-43d4-9de9-36b65001c487.png">
> > Buttons look like links until you hover: <img alt="Screenshot 2023-03-30 at 9 52 28 AM" width="317" src="https://user-images.githubusercontent.com/1116876/228893808-9b0a36a7-7fc7-4caa-8b51-da3df9fea9a8.png"> <img alt="Screenshot 2023-03-30 at 9 52 46 AM" width="404" src="https://user-images.githubusercontent.com/1116876/228893810-2c260365-8c4b-4001-a407-3f35f8626684.png">
> > I understand you have other PRs that extend this, but I need all the elements you are working with here to either look the same or better. I don't want to merge anything that looks worse, even if a different PR adjusts it.
> 
> I will be making the necessary style adjustments in this PR: #295
> 
> Please note that there is a video in the description which demonstrates how I will manage the clear button width and other styles.

What is the name of the theme you are using?

stephywells commented:
@shervElmi It looks like css was removed that was styling these. Here's the css from the current version:
<img width="289" alt="Screenshot 2023-03-30 at 6 43 48 PM" src="https://user-images.githubusercontent.com/1116876/228994888-3405ed7d-160a-4a0b-a847-f41e1c9ab4ea.png">

This class is used with the "Button style. Override WP theme button styling" setting. This PR removes the button bg color.

shervElmi commented:
> @shervElmi It looks like css was removed that was styling these. Here's the css from the current version: <img alt="Screenshot 2023-03-30 at 6 43 48 PM" width="289" src="https://user-images.githubusercontent.com/1116876/228994888-3405ed7d-160a-4a0b-a847-f41e1c9ab4ea.png">
> 
> This class is used with the "Button style. Override WP theme button styling" setting. This PR removes the button bg color.

With these styles, the button design appears unattractive, as shown in the following image:

<img width="760" alt="image" src="https://user-images.githubusercontent.com/69119241/230134043-894f0430-8c77-4297-befe-84545a83be0b.png">

shervElmi commented:
@stephywells There is no padding or border style in the theme for the .button HTML class! So, how can I apply our styles without changing the theme button styles for the button class?

stephywells commented:
> With these styles, the button design appears unattractive,

@shervElmi can you please attach a screenshot of those buttons in the current version with that theme?

We can't remove the buttons styling we've added just because it doesn't look good in one theme. We may need to add an additional class for that theme on the buttons instead. We can see what that theme uses, and add the class if that theme is installed.

shervElmi commented:
> > With these styles, the button design appears unattractive,
> 
> @shervElmi can you please attach a screenshot of those buttons in the current version with that theme?
> 
> We can't remove the buttons styling we've added just because it doesn't look good in one theme. We may need to add an additional class for that theme on the buttons instead. We can see what that theme uses, and add the class if that theme is installed.

### WP 2023 Theme:
<img width="708" alt="image" src="https://user-images.githubusercontent.com/69119241/230199269-6b2fe34d-e434-4e9c-a425-38385fc606a6.png">

### Enfold Theme:
<img width="864" alt="image" src="https://user-images.githubusercontent.com/69119241/230198880-df50e5e4-1436-49e9-945f-5411de4d1359.png">

### Astra Theme:
<img width="1275" alt="image" src="https://user-images.githubusercontent.com/69119241/230199096-d1540944-073d-4742-bcd3-71e9c88d639c.png">

shervElmi commented:
@stephywells, I've committed the necessary changes. I would appreciate it if you could review this PR. Thank you!

stephywells commented on assets/css/less/front-end/elements/forms.less:
```diff
@@ -0,0 +1,161 @@
+		margin: 0 12px 3px 6px;
+		background: @white;
+		color: @color-text;
+		font-size: 12px;
```
Can we set this from a parameter to more easily control and make changes?

stephywells commented on assets/css/less/front-end/elements/buttons.less:
```diff
@@ -0,0 +1,112 @@
+		align-items: center;
+		justify-content: center;
+		cursor: pointer;
+		white-space: nowrap;
```
Looks like we're setting the button padding for some themes like
`.wpbdp-wp-theme-twentytwentytwo .wpbdp-page .wpbdp-main-links .button`

It looks like we need to add this for the 2023 theme too?

stephywells reviewed (changes requested):
Nice! This is looking much better!

I just have a couple small requests left.

shervElmi commented on assets/css/less/front-end/elements/buttons.less:
```diff
@@ -0,0 +1,112 @@
+		border: 1px solid var(--bd-main-color) !important;
+		background-color: var(--bd-main-color) !important;
+		color: @white;
+		text-decoration: none;
```
The user expects our options to always work, and it's important to ensure a consistent user experience. So, I think yes, we can add the `!important` rule. If this solution is not acceptable, please let me know, and I'll remove it instead.

stephywells replied:
Yes, let's add !important. Thanks!

stephywells replied:
This still hasn't been updated.

shervElmi replied:
I can see the `!important!`
<img width="514" alt="image" src="https://user-images.githubusercontent.com/69119241/230513901-5bdf2e4a-7a55-4d7f-b78a-40d4fb4f1271.png">

stephywells replied:
This is for `text-decoration: none;`

shervElmi commented:
@stephywells, I've made updates to the PR based on your feedback. I would greatly appreciate it if you could review this PR again.

stephywells commented on assets/css/less/front-end/abstracts/variables/typography.less:
```diff
@@ -6,6 +6,7 @@
 
 // Font Sizes
 @font-size-alert: 14px;
+@font-size-form-field-with-error: 12px;
```
Maybe we can use something more generic here to make them easier to reuse? Like @font-size-sm and $font-size-xs?

stephywells replied:
It would be nice to update @font-size-alert also to something more easily reusable. like this:
```
@font-size-sm: 14px;
@font-size-xs: 12px;
```

stephywells commented on assets/css/less/wordpress-default-themes.less:
```diff
@@ -106,29 +107,18 @@
 
-		.wpbdp-button,
-		input[type="submit"] {
-			color: var(--wp--preset--color--background);
```
Was there a reason to remove this?

shervElmi replied:
We need a white color for buttons, similar to the following image:
<img width="869" alt="image" src="https://user-images.githubusercontent.com/69119241/230505841-3c8136bb-d8b1-4add-920b-16a7147e9cc6.png">

stephywells replied:
If the setting to override the buttons is off, this css should be used. Can you please attach a screenshot of these buttons in the current version and in this version with the override button setting off?

shervElmi replied:
<img width="813" alt="image" src="https://user-images.githubusercontent.com/69119241/230508177-64d1a8d8-7eaf-481c-a331-55269bb9ee04.png">

shervElmi replied:
![image](https://user-images.githubusercontent.com/69119241/230508224-cd4ecc9e-633a-4aa1-a96b-c89e0358a809.png)

stephywells replied:
Thanks! When that setting is off, the button should have the same colors as before.

shervElmi replied:
<img width="319" alt="image" src="https://user-images.githubusercontent.com/69119241/230511826-db877d68-0aa1-4b0c-b286-9370b7aa1c43.png">

<img width="820" alt="image" src="https://user-images.githubusercontent.com/69119241/230511803-75ffb709-767b-42e7-accd-1ad60b0d840d.png">

---

<img width="340" alt="image" src="https://user-images.githubusercontent.com/69119241/230511893-cecaf015-2ab5-4b73-ba4d-62acc3eb75c0.png">

<img width="710" alt="image" src="https://user-images.githubusercontent.com/69119241/230511880-94eaa3eb-5a0e-4b93-99a3-51b49ef52d05.png">

stephywells replied:
Hmm... It looks like the "Add listing" and "Directory" buttons were shown as links before. Can we fix that too? I don't think it's an improvement to change all of those to buttons.

shervElmi replied:
@stephywells, we have already styled the WP 2022 theme as shown below. Are you suggesting that we only remove button styles for Search Box Links in the WP 2023 theme?

### WP 2022 theme - BD Core 6.3.3 - Disable Button Style:
<img width="775" alt="image" src="https://user-images.githubusercontent.com/69119241/230516929-f7e854b3-8664-49e3-b7b4-14bb61af52f5.png">

### WP 2022 theme - BD Core 6.3.3 -  Enable Button Style:
<img width="729" alt="image" src="https://user-images.githubusercontent.com/69119241/230517025-21cb366b-b72b-4134-9c13-d2cfa56c7813.png">

stephywells replied:
@shervElmi I'm referring to the before screenshot here:
<img width="355" alt="image" src="https://user-images.githubusercontent.com/69119241/230508362-e88b98b9-7531-4880-ab17-9a64f0bca62e.png">

"Directory" and "Add listing" are links there instead of buttons. It feels weird to have all three of those as primary buttons, especially since they are taller than the "find listings" button. In this demo, the buttons are smaller:
https://demo.businessdirectoryplugin.com/

So how about this?
- Remove the "button wpbdp-button" class on all those links except "Add Listing"
- Use a secondary button style for "add listing" (the button color is used for the border and text color with a transparent background. On hover, reverse the colors.)
- Vertically center them using flex and flex gap:
```
.wpbdp-main-links {
display: flex;
justify-content: flex-end;
gap: 12px; (I'm not sure what's the standard gap size)
align-items: center;
}
```

stephywells commented on includes/helpers/functions/templates-ui.php:
```diff
@@ -143,9 +143,10 @@ function( $x ) {
 			$count_str  = apply_filters( 'wpbdp_categories_item_count_str', $count_str, $term );
-			$item_html .= $count_str;
+			$category_text = _n( '%s listing', '%s listings', $count_str, 'business-directory-plugin' );
+			$item_html .= '<span class="wpbdp-category-count">' . sprintf( $category_text, $count_str ) . '</span>';
```
@shervElmi could you point me to the design we are after with this? I'm not quite sure how it should look.

shervElmi replied:
In the meantime, I've added the `wpbdp-block` helper class to our HTML, which currently results in the following appearance:
<img width="660" alt="image" src="https://user-images.githubusercontent.com/69119241/230506647-06c01496-b3ff-4d28-b1f0-8c2daa99a89a.png">

With the upcoming PR, the design is expected to change to something like this:
<img width="439" alt="image" src="https://user-images.githubusercontent.com/69119241/230506736-2536aef1-0e70-4867-8622-b2b7b0723482.png">

shervElmi replied:
Actually, in this [PR](https://github.com/Strategy11/business-directory-categories/pull/7), I have implemented the Figma design, and you can see the result by checking out the video demonstration linked in the PR description:

PR: https://github.com/Strategy11/business-directory-categories/pull/7

Video Demo:
https://www.loom.com/share/8e75240516db4118a9a269c5de2c8050

stephywells replied:
@shervElmi can you check this with the Category images plugin turned off too?

It looks like we need to:
- [ ] set the list style to none (see my screenshot)
- [ ] set the font size bigger for the category name (in your screenshot)
- [ ] remove the underline (both screenshots)

shervElmi replied:
Thank you for your thorough review; I appreciate your attention to detail.

I've reverted the changes to the category count string in the Core plugin and will implement add_filter within the Category Image Plugin to modify the text. I believe this is the right solution.

shervElmi commented:
@stephywells, I think the PR is now prepared for your final review.

stephywells commented on includes/helpers/functions/templates-ui.php:
```diff
@@ -329,7 +329,7 @@ function wpbdp_main_links( $buttons = null ) {
 	if ( in_array( 'create', $buttons, true ) ) {
 		$html[] = '<a href="' . esc_url( wpbdp_url( 'submit_listing' ) ) . '" id="wpbdp-bar-submit-listing-button" class="button wpbdp-button">' .
-			esc_html__( 'Add Listing', 'business-directory-plugin' ) .
+			'+ ' . esc_html__( 'Add Listing', 'business-directory-plugin' ) .
```
I'm not loving the way this looks. I think we removed this since the design was created. Can we revert this line too?

shervElmi commented:
Hi @stephywells,

I've made the necessary changes and the PR is now ready for your review once more.

stephywells reviewed (changes requested):
It's close! Let's fix this:
https://github.com/Strategy11/business-directory-plugin/actions/runs/4597222818 (this link is for a different PR, but it shows the reason for the error)

The last commit also broke PHPstan. I'm not sure why true if being used in a bunch of conditions now.

shervElmi commented:
> It's close! Let's fix this: https://github.com/Strategy11/business-directory-plugin/actions/runs/4597222818 (this link is for a different PR, but it shows the reason for the error)
> 
> The last commit also broke PHPstan. I'm not sure why true if being used in a bunch of conditions now.

My apologies for the confusion! While working on the button elements design, I used 'true' to display all buttons, which unintentionally caused the issue. I'll address this promptly. I'm so sorry!

shervElmi commented:
@stephywells, I apologize for the oversight in the last commit. It has been corrected and is now ready for your review once more. Could you kindly take a look at it again? Thank you.

shervElmi commented:
@stephywells, I understand that you have a busy schedule, but I would greatly appreciate it if you could kindly take some time to review this once more. Thank you in advance for your assistance!

stephywells commented:
@shervElmi I'm not seeing any commits related to my CodeCeption comment. Have you looked into that?

shervElmi commented on .github/workflows/codeception.yml:
@stephywells, I have made updates that successfully address the Codeception errors. Can you please confirm if it's appropriate to use PHP 7.4 with WordPress 5.8?

shervElmi replied:
Thank you for your suggestion, Steph. I have thoroughly tested the functionality with various PHP versions, and it seems to work only with PHP 7.4.

stephywells reviewed (approved):
Yay! It looks ready to merge now. Thank you for all the fixes!
